packages/backend/src/auth-service-client/client.ts

```diff
@@ -0,0 +1,88 @@
+interface Tenant {
+  id: string
+  idpConsentUrl: string
+  idpSecret: string
+}
+
+export class AuthServiceClientError extends Error {
+  constructor(
+    message: string,
+    public status: number,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public details?: any
+  ) {
+    super(message)
+    this.status = status
+    this.details = details
+  }
+}
+
+export class AuthServiceClient {
```

Comment:
Almost feels like this should be a different package outside of `backend` but this is the only consumer at this point so I felt like this keeps it simpler.

Reply:
Intention is to add this as a dep to the tenant service, when thats merged in.